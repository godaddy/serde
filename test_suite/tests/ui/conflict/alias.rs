use serde_derive::Deserialize;

#[derive(Deserialize)]
struct S1 {
    /// Expected error on "alias b", because this is a name of other field
    /// Error on "alias a" is not expected because this is a name of this field
    /// Error on "alias c" is not expected because field `c` is skipped
    #[serde(alias = "a", alias = "b", alias = "c")]
    a: (),

    /// Expected error on "alias c", because it is already used as alias of `a`
    #[serde(alias = "c")]
    b: (),

    #[serde(skip_deserializing)]
    c: (),
}

#[derive(Deserialize)]
struct S2 {
    /// Expected error on "alias c", because this is a name of other field after
    /// applying rename rules
    #[serde(alias = "b", alias = "c")]
    a: (),

    #[serde(rename = "c")]
    b: (),
}

#[derive(Deserialize)]
#[serde(rename_all = "UPPERCASE")]
struct S3 {
    /// Expected error on "alias B", because this is a name of field after
    /// applying rename rules
    #[serde(alias = "B", alias = "c")]
    a: (),
    b: (),
}

fn main() {}
