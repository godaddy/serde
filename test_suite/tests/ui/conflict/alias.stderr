error: alias `b` conflicts with deserialization name of other field
 --> tests/ui/conflict/alias.rs:5:5
  |
5 | /     /// Expected error on "alias b", because this is a name of other field
6 | |     /// Error on "alias a" is not expected because this is a name of this field
7 | |     /// Error on "alias c" is not expected because field `c` is skipped
8 | |     #[serde(alias = "a", alias = "b", alias = "c")]
9 | |     a: (),
  | |_________^

error: alias `c` already used by field a
  --> tests/ui/conflict/alias.rs:11:5
   |
11 | /     /// Expected error on "alias c", because it is already used as alias of `a`
12 | |     #[serde(alias = "c")]
13 | |     b: (),
   | |_________^

error: alias `c` conflicts with deserialization name of other field
  --> tests/ui/conflict/alias.rs:21:5
   |
21 | /     /// Expected error on "alias c", because this is a name of other field after
22 | |     /// applying rename rules
23 | |     #[serde(alias = "b", alias = "c")]
24 | |     a: (),
   | |_________^

error: alias `B` conflicts with deserialization name of other field
  --> tests/ui/conflict/alias.rs:33:5
   |
33 | /     /// Expected error on "alias B", because this is a name of field after
34 | |     /// applying rename rules
35 | |     #[serde(alias = "B", alias = "c")]
36 | |     a: (),
   | |_________^
